### PR TITLE
Add the start of a README.

### PR DESCRIPTION
**Pull request by dlorenc**
See this staged here: https://github.com/dlorenc/distroless/tree/readme#this-repository-contains-source-code-used-to-build-distroless-docker-images

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,67 @@
...
+Follow these steps to get started:
+
+* Pick the right base image for your application stack
+  We publish the following distroless base images on `gcr.io`:
+  <add links to docs>
+
+    * [gcr.io/distroless/base]()
+    * [gcr.io/distroless/java]()
+    * [gcr.io/distroless/java/jetty]()
+    * [gcr.io/distroless/cc]()
```
Comment (mattmoor) [inline README.md]: We haven't published `cc`

File: README.md [hunk 2]
```diff
@@ -1,3 +1,67 @@
...
+  FROM golang:1.8 as build
+
+  WORKDIR /go/src/app
+  COPY . .
+
+  RUN go-wrapper download   # "go get -d -v ./..."
+  RUN go-wrapper install
+
+  # Now copy it into our cc image.
+  FROM gcr.io/distroless/cc
```
Comment (mattmoor) [inline README.md]: For Go, can't I just use `base`?

File: README.md [hunk 3]
```diff
@@ -1,3 +1,67 @@
...
+
+We have some examples on how to run some common application stacks in the /examples directory.
+See here for:
+
+* [Java]()
+  * [Jetty]()
+* [Python]()
+* [Golang]()
+
+See here for exmaples on how to complete some common tasks in your image:
```
Comment (mattmoor) [inline README.md]: typo: exmaples

File: README.md [hunk 4]
```diff
@@ -1,3 +1,67 @@
...
+### Docker
+
+Docker multi-stage builds make using distroless images easy.
+Follow these steps to get started:
+
+* Pick the right base image for your application stack
+  We publish the following distroless base images on `gcr.io`:
+  <add links to docs>
+
+    * [gcr.io/distroless/base]()
```
Comment (mattmoor) [inline README.md]: I think we should probably have a README under each directory that talks about its expected usage (e.g. with java using the jar via CMD, with jetty where to put the webapp).

File: README.md [hunk 5]
```diff
@@ -1,3 +1,67 @@
 [![Build Status](https://travis-ci.org/GoogleCloudPlatform/distroless.svg?branch=master)](https://travis-ci.org/GoogleCloudPlatform/distroless)
 
 # This repository contains source code used to build "distroless" docker images.
```
Comment (mattmoor) [inline README.md]: Let's fit this on a line, maybe just: Distroless Docker Images?

File: README.md [hunk 6]
```diff
@@ -1,3 +1,67 @@
...
+* [Java]()
+  * [Jetty]()
+* [Python]()
+* [Golang]()
+
+See here for exmaples on how to complete some common tasks in your image:
+
+* [Adding and running as a non-root user]()
+* [Including debian packages]()
+* What else?
```
Comment (mattmoor) [inline README.md]: We should anticipate documenting the release process and how to depend on these for stable vs. bleeding edge.

File: README.md (hunk 5 above)
Comment (dlorenc) [inline README.md]: Done.

File: README.md (hunk 4 above)
Comment (dlorenc) [inline README.md]: Yeah, added some placeholders for now.

**Comment by dlorenc** [comment]
TODO:

- [ ] Fill in language-specific READMEs
- [ ] Fill in the "Why" section
- [ ] Fill in the release process section
- [ ] Fill in the ca-certs and nonroot sections

File: README.md [hunk 7]
```diff
@@ -1,3 +1,68 @@
 [![Build Status](https://travis-ci.org/GoogleCloudPlatform/distroless.svg?branch=master)](https://travis-ci.org/GoogleCloudPlatform/distroless)
 
-# This repository contains source code used to build "distroless" docker images.
+# "Distroless" Docker Images
```
Comment (mattmoor) [inline README.md]: nit: Can you move the badge under the header

File: README.md [hunk 8]
```diff
@@ -1,3 +1,68 @@
...
+
+"Distroless" images contain only your application and it's runtime dependencies.
```
Comment (mattmoor) [inline README.md]: its

File: README.md [hunk 9]
```diff
@@ -1,3 +1,68 @@
...
+They do not contain package managers, shells any other programs you would expect to find in a standard Linux distribution.
+
+For more information, see this talk.
```
Comment (mattmoor) [inline README.md]: You can link to this for now, once the video is posted we can link to that:
https://swampup2017.sched.com/event/A6CW/distroless-docker-containerizing-apps-not-vms?iframe=no&w=100%&sidebar=yes&bg=no

File: README.md [hunk 10]
```diff
@@ -1,3 +1,68 @@
...
+
+## Why should I use distroless images?
```
Comment (mattmoor) [inline README.md]: Restricting what's in your runtime to container to precisely what's necessary for your app is a best practice employed by Google and other tech giants that have used containers in production for many years.  It improves the signal to noise of scanners (e.g. CVE) and reduces the burden of establishing provenance to just what you need.

File: README.md [hunk 11]
```diff
@@ -1,3 +1,68 @@
...
+Follow these steps to get started:
+
+* Pick the right base image for your application stack
+  We publish the following distroless base images on `gcr.io`:
+    * [gcr.io/distroless/base](base/README.md)
+    * [gcr.io/distroless/python2.7](python2.7/README.md)
+    * [gcr.io/distroless/java](java/README.md)
+    * [gcr.io/distroless/java/jetty](java/jetty/README/md)
+    * [gcr.io/distroless/cc](cc/README.md)
+* Write a multi-stage docker file.
```
Comment (mattmoor) [inline README.md]: I'd add a note that this required 17.05 or later

File: README.md [hunk 12]
```diff
@@ -1,3 +1,68 @@
...
+  RUN go-wrapper download   # "go get -d -v ./..."
+  RUN go-wrapper install
+
+  # Now copy it into our cc image.
+  FROM gcr.io/distroless/base
+  COPY --from=build /go/bin/app /
+  CMD ["/app"]
+  ```
+
+### Bazel
```
Comment (mattmoor) [inline README.md]: We should just link to `bazelbuild/rules_docker`, which I should update to point to these :)

File: README.md [hunk 13]
```diff
@@ -1,3 +1,68 @@
...
+<show docker_build usage>
+
+## Examples
+
+We have some examples on how to run some common application stacks in the /examples directory.
+See here for:
+
+* [Java]()
```
Comment (mattmoor) [inline README.md]: You should be able to populate these (mostly), right?

File: base/README.md [hunk 14]
```diff
@@ -0,0 +1,16 @@
+# Documentation for `gcr.io/distroless/base`
+
+## Image Contents
+
+This image contains a minimal Linux, glibc-based system. It is intended for use directly by "mostly-statically compiled" languages like Go, Rust or D.
```
Comment (mattmoor) [inline base/README.md]: Rust and D use "cc"

Comment (mattmoor) [inline base/README.md]: I don't see a cc/README.md?

File: java/README.md [hunk 15]
```diff
@@ -0,0 +1,13 @@
...
+
+This image contains a minimal Linux, openjdk-based runtime.
+
+Specifically, the image contains everything in the [base image](../base/README.md), plus:
+
+* openjdk8 and it's dependencies.
+
+## Usage
+
+The entrypoint of this image is set to the equivalent of "java -jar", so this image expects users to supply a path to a JAR file in the CMD.
```
Comment (mattmoor) [inline java/README.md]: TODO: Dockerfile example
TODO: docker_build example

Comment (dlorenc) [inline java/README.md]: Added to the TODO comment above.

File: base/README.md (hunk 14 above)
Comment (dlorenc) [inline base/README.md]: Yeah sorry, need to rebase on yours CC split.

File: README.md (hunk 13 above)
Comment (dlorenc) [inline README.md]: Done.

File: README.md (hunk 12 above)
Comment (dlorenc) [inline README.md]: Done.

File: README.md (hunk 11 above)
Comment (dlorenc) [inline README.md]: Done.

File: README.md (hunk 10 above)
Comment (dlorenc) [inline README.md]: Nice :)

File: README.md (hunk 9 above)
Comment (dlorenc) [inline README.md]: Done.

File: README.md (hunk 8 above)
Comment (dlorenc) [inline README.md]: Done.

File: README.md (hunk 7 above)
Comment (dlorenc) [inline README.md]: Done.

File: README.md [hunk 16]
```diff
@@ -1,3 +1,76 @@
...
+  # Start by building the application.
+  FROM golang:1.8 as build
+
+  WORKDIR /go/src/app
+  COPY . .
+
+  RUN go-wrapper download   # "go get -d -v ./..."
+  RUN go-wrapper install
+
+  # Now copy it into our cc image.
```
Comment (mattmoor) [inline README.md]: stale comment: `cc`

File: cc/README.md [hunk 17]
```diff
@@ -0,0 +1,13 @@
...
+
+This image contains a minimal Linux, glibc runtime for "mostly-statically compiled" languages like Rust and D.
+
+Specifically, the image contains everything in the [base image](../base/README.md), plus:
+
+* libgcc1 and its dependencies.
+
+## Usage
+
+Users are expected to include their compiled application and set the correct cmd in their image.
```
Comment (mattmoor) [inline cc/README.md]: nit: you capitalize `CMD` elsewhere

**Pull request merged**